src/components/list/PagedList.js

```diff
@@ -0,0 +1,181 @@
+import { useRect } from '@reach/rect'
+import { ChevronLeft, ChevronRight } from '@styled-icons/boxicons-regular'
+import { useEffect, useRef, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components/macro'
+
+import { useMap } from '../../contexts/MapContext'
+import { getLocations } from '../../utils/api'
+import Checkbox from '../ui/Checkbox'
+import LoadingIndicator from '../ui/LoadingIndicator'
+import SquareButton from '../ui/SquareButton'
+import EntryList from './EntryList'
+
+const LIMIT = 30
+
+const StyledContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+`
+
+const StyledListContainer = styled.div`
+  height: 100%;
+`
+
+const StyledPageInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
+`
+
+const StyledPageNav = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: 10px;
+  border-top: 1px solid ${({ theme }) => theme.secondaryBackground};
+`
+
+const StyledNavButtonContainer = styled.div`
+  display: flex;
+  button {
+    &:not(:last-child) {
+      margin-right: 4px;
+    }
+  }
+`
+
+const StyledCheckboxContainer = styled.div`
+  display: flex;
+  padding: 10px;
+  color: ${({ theme }) => theme.secondaryText};
+`
+
+const PagedList = () => {
+  const history = useHistory()
+  const container = useRef()
+  const rect = useRect(container) ?? { width: 0, height: 0 }
+  const { view } = useMap()
+  const [locations, setLocations] = useState([])
+  const [currentOffset, setCurrentOffset] = useState(0)
+  const [totalLocations, setTotalLocations] = useState(0)
+  const [updateOnMapMove, setUpdateOnMapMove] = useState(false)
+  const [currentView, setCurrentView] = useState(undefined)
```

Comment:
You might want to add a comment here stating that currentView is necessary because we need to copy it over when update list on map move is unchecked.